src/Features/LanguageServer/Protocol/Handler/RequestTelemetryLogger.cs

```diff
@@ -92,6 +94,14 @@ public void Dispose()
             return;
         }
 
+        // Flush all telemetry logged through TelemetryLogging
+        TelemetryLogging.Flush();
+
+        TelemetryLogging.Flushed -= OnFlushed;
+    }
+
+    private void OnFlushed(object? sender, EventArgs e)
```

Comment:
I don't remember if the queries are resilient to multiple events for the same server for the same session.  I assume they are since I don't think we drill down into the session in particular, but I can't remember.

Reply:
The queries that I know don't drill into the session, so they are resilient. If we find some that do use the session, it seems feasible to change them to allow multiple items from a session.